### ISSUE TITLE
Upload binary to github

### ISSUE THREAD

**Issue by bn3t** [chasinglogic/cargo-hublish#3]
Hi,
I could use your tool to make a release but I was kind of hoping that it would upload the binary. Instead, it uploads the source but github does that already automatically when a tag is made.
Thanks for your help,
Bernard.

**Comment by chasinglogic** [comment]
So this is what I opened issue #1 about when I created this project. There are multiple problems which I don't think have easy answers:

- What do we do if there isn't a release build?
- How do we know if it has actually been compiled against the tagged commit?
 - What do we do if we think it hasn't been?
- How do we know what else should be in the tarball? Presumably you want to at least ship the LICENSE file and any docs.
- What  should the resulting tarball be named?
- How do we handle multiple platforms / cross-compiling
- Do we do zips for everything since it's supported everywhere or do tarballs for Unix and zips for WIndows

I haven't been writing much Rust tooling lately so I haven't found what I think the answer to all of these questions should be. However, I believe they will need answer before they get implemented in this tool.

**Comment by bn3t** [comment]
Thanks for your reply. These are all valid points.
In the meanwhile, I found a solution to my problem. I needed a way to build and macOS tool and I thought that travis-ci could do that on the free plan (normally macOS builds are pay feature). Using the tool described in https://github.com/japaric/trust it's possible to cross compile rust to darwin. This is exactly what I needed. japaric/trust handles the cross compilation and travis-ci does the release to GitHub.
You can see the result of a release done via travis-ci [here](https://github.com/bn3t/screenresolution-rs/releases/tag/v0.1.6).
In conclusion, I don't think I will need cargo-hublish anymore :-(.

**Comment by chasinglogic** [comment]
No problem, glad you found a solution to your problem. Thanks for the bug report
